tests/component/encapsulation.cpp

```diff
@@ -22,6 +22,8 @@ limitations under the License.
 
 #include <iostream>
 
+/*
+// TODO Removed this test as doesn't actually test anything ... except show a bug in how we deal with parents
```

Comment:
I actually think this should be here but turned into a proper test, it does show how the library behaves which is important.

Reply:
But the issues it raises are being addressed in another branch

Reply:
... so the test should go there instead maybe?

Reply:
Have updated the comments, and we can reinstate the test when we're ready to address the parent stuff.